packages/block-editor/src/components/block-patterns-list/index.js

```diff
@@ -199,23 +221,49 @@ function BlockPatternsList(
 	},
 	ref
 ) {
-	const [ activeCompositeId, setActiveCompositeId ] = useState( undefined );
+	const [ activePattern, setActivePattern ] = useState( null ); // State to track active pattern
 
 	useEffect( () => {
 		// Reset the active composite item whenever the available patterns change,
 		// to make sure that Composite widget can receive focus correctly when its
 		// composite items change. The first composite item will receive focus.
-		const firstCompositeItemId = blockPatterns.find( ( pattern ) =>
-			shownPatterns.includes( pattern )
-		)?.name;
-		setActiveCompositeId( firstCompositeItemId );
-	}, [ shownPatterns, blockPatterns ] );
+		if ( typeof window !== 'undefined' && window.localStorage ) {
+			// eslint-disable-next-line no-undef
+			const storedPatternName = localStorage.getItem( 'savedPattern' );
 
+			if (
+				storedPatternName &&
+				shownPatterns.some(
+					( pattern ) => pattern.name === storedPatternName
+				)
+			) {
+				// If there's a saved pattern and it exists in shownPatterns, set it as active
+				setActivePattern( storedPatternName );
+			} else {
+				const firstCompositeItemId = blockPatterns.find( ( pattern ) =>
+					shownPatterns.includes( pattern )
+				)?.name;
+
+				if ( firstCompositeItemId ) {
+					setActivePattern( firstCompositeItemId );
+				}
+			}
+		}
+	}, [ shownPatterns, blockPatterns ] );
+	const handleClickPattern = ( pattern ) => {
+		// Check if we are in a browser environment and localStorage is available
+		if ( typeof window !== 'undefined' && window.localStorage ) {
+			// eslint-disable-next-line no-undef
+			localStorage.setItem( 'savedPattern', pattern.name ); // Save the selected pattern in localStorage
+		}
```

Comment:
I'm not an expert on best practices to save/restore data, but I don't think we usually use `localStorage`. I believe we should use data stores, but @ntsekouras @youknowriad @ellatrix may be able to give more tailored advice.

Reply:
Thanks for the suggestion @ciampo 

I have updated the code with `sessionStorage` since it is using in various scenarios. Please correct me if I'm 
following something wrong.

If I still need to handle this with data stores kind of handling please let me know.

Reply:
Can we avoid using local or session storage entirely? Ideally we would either save this information in local react state (without persisting it to the browser), or otherwise retrieve this information from a data store.
The folks that I pinged above should be able to give more info on this

Reply:
Hi @ciampo 

Now  I have removed the use of local/session storage for storing the selected pattern across page reloads after saving the template. Now, the selected pattern is managed using local React state. This ensures that the selection persists after saving the template but resets when the page is reloaded, even if the template has been saved.

I think this needs further discussion to determine if we should maintain the selected item across page reloads. I'm open to feedback and suggestions from the team regarding this approach.

Looking forward to input from others, and I will make updates based on that.

Thanks!

Reply:
Thank you for your PR @benazeer-ben!


I should start by saying that the suggestions in that `Design` panel are just patterns that are 'connected' to the current active template with the [templateTypes](https://developer.wordpress.org/news/2023/04/13/using-template-patterns-to-build-multiple-homepage-designs/) prop. 

What this means in practice that when we click on one of the suggestions, the current template's content is only updated and it's just blocks without having any information about patterns etc.. So on refresh there isn't a way to tell if the content was actually inserted before with that mechanism.

There is a similar use case with patterns that we add some metadata in the top level block and use that to perform some actions like shuffle, etc.. That has already some limitations because it also expects these patterns to have a single wrapping block.

Having said that, while we could explore a similar solution here, I'm not sure we should. I think it would be still fragile and dependant on the provided templates and its usage is probably a lot less than the patterns case. I don't have strong opinion though for that and if more folks believe we need that info, let's try it.

Reply:
Thank you, Nik!

I propose that we don't try to implement a solution for the time being. And if we even _really_ feel the need for this feature, we should look at it holistically, ie. implementing it in a way that works across all places in the editor.